DeezyMatch/combineVecs.py

```diff
@@ -98,29 +106,32 @@ def combine_vecs(input_file_path="default", qc_modes=["q", "c"], rnn_passes=["fw
             for i, lfile in enumerate(list_files): 
                 if i % print_every == 0: print("%07i" % i, lfile)
                 if len(vecs_ids) == 0:
-                    vecs_ids = torch.load(f"{lfile}", map_location=dl_inputs['general']['device'])
+                    vecs_ids = torch.load(f"{lfile}", map_location=sel_device)
                 else:
-                    vecs_ids = torch.cat((vecs_ids, torch.load(f"{lfile}", dl_inputs['general']['device'])))
+                    vecs_ids = torch.cat((vecs_ids, torch.load(f"{lfile}", sel_device)))
             print()
             # Save combined IDs
             torch.save(vecs_ids, path_id_combined)
+            del vecs_ids
             
-            # Save strings for the first column in queries/candidates files
-            mydf = pd.read_pickle(pathdf)
-            vecs_items = mydf['s1_unicode'].to_numpy()
-            np.save(path_items_combined, vecs_items)
+            if not no_query_candidate_df:
+                # Save strings for the first column in queries/candidates files
+                mydf = pd.read_pickle(pathdf)
```

Comment:
Hi @kasra-hosseini, if the `input_scenario` does not exist as a folder you'll get this error:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/deezy_datadrive/mariona/DeezyMatch/DeezyMatch/combineVecs.py", line 119, in combine_vecs
    mydf = pd.read_pickle(pathdf)
  File "/data/anaconda/envs/py37torch/lib/python3.7/site-packages/pandas/io/pickle.py", line 182, in read_pickle
    return pickle.load(f)
ValueError: unsupported pickle protocol: 5
```

Maybe we could print a warning that the input scenario does not exist.

Reply:
Hi @mcollardanuy, Many thanks for the review. I will go through them one by one. Here is the commit that should address this isse: https://github.com/Living-with-machines/DeezyMatch/commit/f86ab202facfd6a213dc117a8be70d2093eaa4be

Reply:
Perfect, thanks!